docs/admin/dns.md

```diff
@@ -22,17 +24,162 @@ in namespace `bar` can look up this service by simply doing a DNS query for
-in-memory lookup structures to service DNS requests. The dnsmasq container adds DNS caching to improve
-performance. The healthz container provides a single health check endpoint while performing dual healthchecks
-(for dnsmasq and kubedns).
+#### A records
+
+"Normal" (not headless) Services are assigned a DNS A record for a name of the
+form `my-svc.my-namespace.svc.cluster.local`.  This resolves to the cluster IP
+of the Service.
+
+"Headless" (without a cluster IP) Services are also assigned a DNS A record for
+a name of the form `my-svc.my-namespace.svc.cluster.local`.  Unlike normal
+Services, this resolves to the set of IPs of the pods selected by the Service.
+Clients are expected to consume the set or else use standard round-robin
+selection from the set.
+
+### SRV records
+
+SRV Records are created for named ports that are part of normal or Headless
+Services.
+For each named port, the SRV record would have the form
+`_my-port-name._my-port-protocol.my-svc.my-namespace.svc.cluster.local`.
+For a regular service, this resolves to the port number and the CNAME:
+`my-svc.my-namespace.svc.cluster.local`.
+For a headless service, this resolves to multiple answers, one for each pod
+that is backing the service, and contains the port number and a CNAME of the pod
+of the form `auto-generated-name.my-svc.my-namespace.svc.cluster.local`.
+
+### Backwards compatibility
+
+Previous versions of kube-dns made names of the for
+`my-svc.my-namespace.cluster.local` (the 'svc' level was added later).  This
+is no longer supported.
+
+### Pods
+
+#### A Records
+
+When enabled, pods are assigned a DNS A record in the form of `pod-ip-address.my-namespace.pod.cluster.local`.
+
+For example, a pod with ip `1.2.3.4` in the namespace `default` with a dns name of `cluster.local` would have an entry: `1-2-3-4.default.pod.cluster.local`.
+
+
+#### A Records and hostname based on Pod's hostname and subdomain fields
+
+Currently when a pod is created, its hostname is the Pod's `metadata.name` value.
+
+With v1.2, users can specify a Pod annotation, `pod.beta.kubernetes.io/hostname`, to specify what the Pod's hostname should be.
+If the annotation is specified, the annotation value takes precendence over the Pod's name, to be the hostname of the pod.
+For example, given a Pod with annotation `pod.beta.kubernetes.io/hostname: my-pod-name`, the Pod will have its hostname set to "my-pod-name".
+
+With v1.3, the PodSpec has a `hostname` field, which can be used to specify the Pod's hostname. This field value takes precedence over the
+`pod.beta.kubernetes.io/hostname` annotation value.
+
+v1.2 introduces a beta feature where the user can specify a Pod annotation, `pod.beta.kubernetes.io/subdomain`, to specify what the Pod's subdomain should be.
+If the annotation is specified, the fully qualified Pod hostname will be "<hostname>.<subdomain>.<pod namespace>.svc.<cluster domain>".
```

Comment:
Passive voice.

If the annotation is specified -> If you specify the annotation